crates/fuel-core/src/schema/message.rs

```diff
@@ -52,10 +53,13 @@ impl Message {
     async fn da_height(&self) -> U64 {
         self.0.da_height.as_u64().into()
     }
+}
 
-    async fn fuel_block_spend(&self) -> Option<U64> {
-        self.0.fuel_block_spend.map(|v| v.into())
-    }
+#[derive(Enum, Copy, Clone, Eq, PartialEq)]
+#[graphql(remote = "entities::message::MessageStatus")]
+pub enum MessageStatus {
```

Comment:
You can reuse `CoinStatus` because during the next iteration we will move it into coin=)

Reply:
not true, there will be DepositCoins and regular old messages. Not all messages will become coins, just the ones that only include gas.